Added support for pi2

tylerwowen opened the pull request:
I added support for Pi2. Only tested on Pi2. It works on Pi2, but I'm not sure if it is fully working.

tylerwowen committed:
Read cpu info to set base for Pi 2 (+4 squashed commits)
Squashed commits:
[743c229] Read cpu info to set base for Pi 2
[5c70d6a] Read cpu info to set base for Pi 2
[a372e3b] Read cpu info to set base for Pi 2
[d1bbbc4] Read cpu info to set base for Pi 2

Commit:
encode in UFT-8 (+5 squashed commits)
Squashed commits:
[0511bc4] merged from upstream
[05245b5] Added Pi2 support
[ea8f276] Added Pi 2 support
[82a5d92] Added support for Pi2
[5a6727c] Added Pi 2 support

Read cpu info to set base for Pi 2

Read cpu info to set base for Pi 2

Read cpu info to set base for Pi 2

Read cpu info to set base for Pi 2

metachris commented:
Thank you! I will get a Pi 2 soon and then I'll test this PR immediately.

tylerwowen commented:
You are welcome. However, please notice that the PWM part is still not working on Pi 2. I had difficulties to fix that part.

metachris commented:
Alright. Perhaps its worth to take a look at the ServoBlaster library, which was the starting point for the RPIO PWM module:

* https://github.com/richardghirst/PiBits/tree/master/ServoBlaster
* https://github.com/richardghirst/PiBits/issues/43

metachris commented:
Also:

* https://github.com/metachris/RPIO/network
* https://github.com/mw46d/RPIO/commit/a92a859848e3e87f5a6f51614093c2c3cd7d283e

fjbelchi commented:
Any progress with PWM support for Pi 2?

Commit:
Added Pi 2 support

Added support for Pi2

Added Pi 2 support

Added Pi2 support

merged from upstream

encode in UFT-8

WORMSS commented:
Any progress with PWM support for Pi 2? Whole reason I got the Pi2 because I was told it's better than Arduino. But I got the Arduino running a simple PWM within 5 minutes. I've been on this for an hour on the Pi2 trying libraries left right and centre. :bangbang:

jashort commented:
@WORMSS I had good luck with the PWM support in the fork @tylerwowen made. I installed it from https://github.com/tylerwowen/RPIO (It's this pull request, it just hasn't been merged yet)

tylerwowen commented:
@WORMSS Raspberry Pi is not better for PWM. This is from the [official GPIO module documentation](https://pypi.python.org/pypi/RPi.GPIO)

> Note that this module is unsuitable for real-time or timing critical applications. This is because you can not predict when Python will be busy garbage collecting. It also runs under the Linux kernel which is not suitable for real time applications - it is multitasking O/S and another process may be given priority over the CPU, causing jitter in your program. If you are after true real-time performance and predictability, buy yourself an Arduino http://www.arduino.cc !

Use the official library instead, it has software PWM supports.

@jashort Did it really work for PWM? I don't think I could fix it.

WORMSS commented:
@jashort @tylerwowen thank you. I guess I will do a combination of the two then. As I think I will still need certain things that a Pi2 can offer such as nodejs (and ssh if things get really bad). So maybe a Pi2 that triggers an Arduino into action. Fingers crossed. Again, thank  you.

Xfel commented:
Any progress on this? As far as I see the PWM stuff does work now in ServoBlaster after this commit: https://github.com/richardghirst/PiBits/commit/96014c804d68677baf4fff25178f9b152c1f31e6

So could this be adapted for this project too?

metachris commented:
I wasn't aware, thanks for the heads up! If you have time to port it over,
you are more than welcome to :-)
On Nov 11, 2015 15:37, "Xfel" <notifications@github.com> wrote:

> Any progress on this? As far as I see the PWM stuff does work now in
> ServoBlaster after this commit: richardghirst/PiBits@96014c8
> <https://github.com/richardghirst/PiBits/commit/96014c804d68677baf4fff25178f9b152c1f31e6>
>
> So could this be adapted for this project too?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/metachris/RPIO/pull/56#issuecomment-155799127>.
>

cmdwiggles commented:
Any progress with this?
Thoroughly dissatisfied with GPIO.PWM and its ideas of 'pwm', veeerrrrry jittery outputs dont end well when mixed with drones

fukatani committed:
Enable to set GPIO.ALT0.

alexellis commented:
@Xfel @metachris I can test any merge that is done from servo blaster.

If the pwm side of things is based upon the work of servoblaster - how difficult would it be to merge? Was there a lot of change to the methods/interface?

maleadt commented:
Differences between [servoblaster](https://github.com/richardghirst/PiBits/blob/master/ServoBlaster/user/servod.c) and [RPIO](https://github.com/metachris/RPIO/blob/master/source/c_pwm/pwm.c) are significant, so merging isn't a quick and easy job.

I tried porting the PWM part to RPi2, but couldn't get it to work either (it will require more than just updating the peripheral base offset). That said, `servoblaster` didn't really work on my Pi2: initially (first 5 seconds or so) it PWMs fine, but then the board powers-off abruptly. No oops or nothing, so I didn't try to debug. Not sure whether it's specific to my set-up or board though.
So for anybody trying to port the servoblaster Pi2 support over: make sure servoblaster itself works first.

lurch commented on source/c_gpio/py_gpio.c~:
```diff
@@ -0,0 +1,577 @@
+/*
```
This is a backup file (ends with ~ ) so should be deleted from this PR.

lurch commented on source/c_gpio/cpuinfo.c:
```diff
@@ -70,8 +71,10 @@ get_cpuinfo_revision(char *revision_hex)
-        // We'll call Model B+ (0010) rev3
+    } else if (strcmp(revision_hex, "0010") == 0
+							 || strcmp(revision_hex, "a21041") == 0
+							 || strcmp(revision_hex, "a01041") == 0 ) {
```
Looks like the indentation went a bit funny? ;)

lurch commented on source/c_gpio/c_gpio.c:
```diff
@@ -79,8 +100,16 @@ setup(void)
+        gpio_map = (uint32_t *)mmap( (void *)gpio_mem, BLOCK_SIZE, PROT_READ|PROT_WRITE, MAP_SHARED|MAP_FIXED, mem_fd, BCM2708_PERI_BASE + 0x200000);
+		}
+    else{
+				gpio_map = (uint32_t *)mmap( (void *)gpio_mem, BLOCK_SIZE, PROT_READ|PROT_WRITE, MAP_SHARED|MAP_FIXED, mem_fd, BCM2709_PERI_BASE + 0x200000);
```
Strange indentation problems

Commit:
clean up
